[P036] CHG: Header content changed only once within display interval

uwekaditz opened the pull request:
CHG: Header content changed only once within display interval instead every second
CHG: more predefined constants to understand the meaning of the values

uwekaditz committed:
[P036] CHG: Header content only changed once within display interval

CHG: Header content only changed once within display interval. before every second
CHG: more predefined constants to understand the meaning of the values

TD-er commented on src/_P036_FrameOLED.ino:
```diff
@@ -780,16 +787,15 @@ void display_time() {
   display->setColor(BLACK);
-//  display->fillRect(0, 0, 128, 13); // Underscores use a extra lines, clear also.
-  display->fillRect(0, TopLineOffset, 128, 12);   // don't clean line under title.
+  display->fillRect(0, TopLineOffset, 128, 12);   // don't clear line under title.
```
Is this 128 a given for all these controllers, or should this also be P36_MaxDisplayWidth ?

TD-er commented on src/_P036_FrameOLED.ino:
```diff
@@ -798,7 +804,7 @@ int left = 24;
   display->setFont(ArialMT_Plain_16);
   display->setColor(BLACK);
-  display->fillRect(0, 13+TopLineOffset, 128, 64);
+  display->fillRect(0, 13+TopLineOffset, 128, P36_MaxDisplayHeight);
```
Same for this 128 value, must it be P36_MaxDisplayWidth or is it a different value?

uwekaditz commented:
I'm going to change the 128 value to P36_MaxDisplayWidth.
I was checking if the different controllers have different Max_width but both of them are created with a width of 128 pixels.
Each call display->fillRect() with more than 128 will return without doing anything!

uwekaditz committed:
[P036] Value 128 change to P36_MaxDisplayWidth

TD-er commented:
Also with the SH1106?

uwekaditz commented:
The change was general change for all controllers in PLUGIN_TEN_PER_SECOND and in display_header()

uwekaditz committed:
[P036] Commands (PLUGIN_WRITE) reviewed completely

1. Display commands:
oledframedcmd display [on, off, low, med, high]
turns display on or off or changes contrast (low, med or high)
2. Content commands:
oledframedcmd <line> <text> <mode>
displays <text> in line (1-12), next frame is showing the changed line, mode not implemented yet

TD-er commented on src/_P036_FrameOLED.ino:
```diff
@@ -600,30 +621,74 @@ boolean Plugin_036(byte function, struct EventStruct *event, String& string)
+            // content functions
+            success = true;
+            String NewContent = parseStringKeepCase(string, 3);
+            P036_displayLines[LineNo-1] = NewContent;
```
You allow LineNo == P36_Nlines+1

So `P036_displayLines[LineNo-1]` can become `P036_displayLines[P36_Nlines]`, which is out of bounds

Suggestion:
`else if ((LineNo > 0) && (LineNo <= P36_Nlines))`

uwekaditz replied:
Done!

uwekaditz commented:
Thanks, you're very observant.

uwekaditz committed:
[P036] Commands (PLUGIN_WRITE) reviewed completely

Out of bounds in PLUGIN_WRITE
Warnings resolved
